src/app/ClusterObjectUtils.h

```diff
@@ -0,0 +1,312 @@
+/*
```

Comment:
Please avoid usage of names like utils/helpers/support/... . 

In this case 'ClusterObjectUtils' does not help me understand what of cluster object goes in here. Looking at the contents, this seems to contain encode, decode and iterable list. 

please create 'ClusterObjectEncoding.h' for encode/decode and IterableList.h for the list item (or similar naming).

Do not allow a file named global/utils/helpers be a all-in-one bag for everything.

Reply:
I had originally considered `ClusterObjectEncoding.h` as well, but that implies only encoding, when both encode/decode are covered here.

Suggestion: `ClusterObjectSerialization.h`?

Will move `IteratableList` out to a 'ClusterObjectTypes.h' (since I suspect we may grow this set of types in the future), this file becomes just about utility functions for encoding/decoding.

Reply:
Serialization is fine. Or I may even suggest Encoding.h and Decoding.h separately to prioritize readability over typing speed.